core/src/main/java/nu/yona/server/subscriptions/service/UserService.java

```diff
@@ -105,54 +115,63 @@ static User findUserByMobileNumber(String mobileNumber)
 		return userEntity;
 	}
 
-	private UserDTO handleUserSignupUponBuddyRequest(String password, UserDTO userResource, User originalUserEntity)
+	@Transactional
+	public UserDTO updateUserCreatedOnBuddyRequest(UUID id, String tempPassword, UserDTO userResource)
 	{
-		UpdatedEntities updatedEntities = updateUserWithTempPassword(userResource, originalUserEntity);
-		return saveUserWithDevicePassword(password, updatedEntities);
+		User originalUserEntity = getEntityByID(id);
+		if (!originalUserEntity.isCreatedOnBuddyRequest())
+		{
+			throw new YonaException("User is not created on buddy request");
```

Comment:
Throw ``IllegalArgumentException``, as this is a programming error, and change the text into something like this: "Can only update 'user created on buddy request' through this method".